### ISSUE TITLE
Data underfitting of TFT model

### ISSUE THREAD

**Issue by miteshyadav** [jdb78/pytorch-forecasting#523]
I'm running TFT on a custom dataset and noticed that the predictions are under fiyting even after training for 500 epochs. I tried adding more hidden layers and experimenting with different learning rates, but that didn't help either. DeepAR seems to do a good job in predicting patterns, but Im surprised to see TFT not performing that well. Did anyone else has been facing similar issue? Is there any other way to solve this? Thanks

![image](https://user-images.githubusercontent.com/31832306/119691533-e4f14e00-be07-11eb-85bc-f72370ee3d36.png)

**Comment by jdb78** [comment]
How are you normalizing? Use the `GroupNormalizer()` as `target_normalizer` and you should be able to overfit. Because of the forward-looking nature of time series, normalization with an `EncoderNormalizer()` can lead to an inability to dramatically overfit.

**Comment by miteshyadav** [comment]
Thanks for the reply. I am using the GroupNormalizer() as the target normalizer, it underfits despite that. Any other parameter that I should tune for it to overfit?

**Comment by jdb78** [comment]
Could you share your definition of the `TimeSeriesDataSet`?

**Comment by miteshyadav** [comment]
@jdb78 Sure. Here you go

' ' '
from pytorch_forecasting.data.encoders import NaNLabelEncoder
from pytorch_forecasting.data import NaNLabelEncoder


max_prediction_length = 365
max_encoder_length = 24
training_cutoff = df["time_idx"].max() - max_prediction_length

training = TimeSeriesDataSet(
    df[lambda x: x.time_idx <= training_cutoff],
    time_idx="time_idx",
    target="y",
    categorical_encoders={"steam_id": NaNLabelEncoder().fit(df.steam_id)},
    group_ids=["steam_id"],
    min_encoder_length=max_encoder_length // 2,  # keep encoder length long (as it is in the validation set)
    max_encoder_length=max_encoder_length,
    min_prediction_length=1,
    max_prediction_length=max_prediction_length,
    static_categoricals=["steam_id"] ,
    static_reals=['review_score', 'total_positive','total_negative', 'total_reviews'] + extracted_features.columns.tolist()[2:],
    time_varying_known_categoricals=["month","season_sale","promo_flag"],
    variable_groups={"season_sale": season_sale} , # group of categorical variables can be treated as one variable
    time_varying_known_reals=["time_idx", "price","discount"],
    time_varying_unknown_categoricals=[],
    time_varying_unknown_reals=[
        "y",
        "log_y"
    ],
    target_normalizer=GroupNormalizer(
        groups=["steam_id"], transformation="softplus"
    ),  # use softplus and normalize by group
    add_relative_time_idx=True,
    add_target_scales=True,
    add_encoder_length=True,
    allow_missings=True,
)

' ' '

**Comment by jdb78** [comment]
Are you trying to overfit on a very small subset of the data? See https://pytorch-lightning.readthedocs.io/en/stable/common/trainer.html#overfit-batches

**Comment by DeastinY** [comment]
Hey, even with the overfit_batches setup I don't seem to get TFT to overfit either. Loss goes down to 0.001 on train and validation, val_mae on tensorboard is also crazy low, but when I test like this, I still get a fit that's not even close to overfitting.

```
actuals = torch.cat([y[0] for x, y in iter(val_dataloader)])
predictions = model.predict(val_dataloader)
mae = (actuals.reshape(-1, predictions.shape[1]) - predictions).abs().mean().item()
print(mae)
```

```
raw_predictions, x = model.predict(val_dataloader, mode="raw", return_x=True)
fig = model.plot_prediction(x, raw_predictions, idx=0, add_loss_to_title=True)
```
